plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.ts

```diff
@@ -174,13 +177,35 @@ export class BitbucketServerEntityProvider implements EntityProvider {
   }
 
   private async findEntities(): Promise<Entity[]> {
+    this.logger.info('Finding entities in Bitbucket Server repositories');
+
     const client = BitbucketServerClient.fromConfig({
       config: this.integration.config,
     });
     const projects = paginated(options =>
       client.listProjects({ listOptions: options }),
     );
     const result: Entity[] = [];
+
+    let parseFunction:
+      | ((opt: ParseOptions) => Promise<AsyncIterable<Entity>>)
+      | undefined;
+
+    if (this.config.throttling) {
+      const throttle = pThrottle({
+        limit: this.config.throttling.count,
+        interval: durationToMilliseconds(this.config.throttling.interval),
+      });
+
+      parseFunction = throttle((opt: ParseOptions) => {
+        return this.parse(opt);
```

Comment:
I'm not sure but, is this really throttling what it should throttle? The default parse function doesn't do anything at all, and if it were only about the parse function then the caller could be responsible themselves for passing in an already throttled callback. And what if that callback performs half a million requests per call, how does that affect the throttling?

Isn't it the case that, what we want to throttle is really all `client` requests that we make? Maybe I misunderstand, but this looked surprising to me.

Reply:
It's my understanding that part of the parse function is that it makes a hit to the Bitbucket Server API to look at the configured url for matching entities. I calls to list the repositories, but this seems to be the place where the actual lookup against files in bitbucket happens.

Reply:
I understand better now. I have moved the throttling to be around the `fetch` function in the client and url readers